File: A2/Sem4/MPP/Teme/TemaCsNetworkingJson/services/IBileteServices.cs
using bilete.model;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace bilete.services
{
    public interface IBileteServices
    {
        void login(User user, IBileteObserver client);
        void logout(User user, IBileteObserver client);
        Show[] findAllArtisti();
        Show[] findShowsOnDate(DateOnly date);
        void sellTicket(Ticket ticket);
    }
}

File: A2/Sem4/MPP/Teme/BileteJavaSocketsJson/BileteServices/src/main/java/bilete/services/IBileteServices.java
package bilete.services;

import bilete.domain.Show;
import bilete.domain.Ticket;
import bilete.domain.User;

import java.time.LocalDate;

public interface IBileteServices {
    void login(User user, IBileteObserver client) throws BileteException;
    void logout(User user, IBileteObserver client) throws BileteException;
    Show[] findAllArtisti() throws BileteException;
    Show[] findShowsOnDate(LocalDate date) throws BileteException;
    void sellTicket(Ticket ticket) throws BileteException;
}
